finetune/base_models/long_doc/featurizer.py

```diff
@@ -0,0 +1,72 @@
+import tensorflow as tf
+
+from finetune.base_models.gpt.featurizer import dropout, embed
+from finetune.base_models.tcn.featurizer import TemporalBlock
+
+
+def long_doc_featurizer(
+    X,
+    encoder,
+    config,
+    train=False,
+    reuse=None,
+    lengths=None,
+    **kwargs
+):
+    """
+    The featurizer element of the finetuning model. Maps from tokens ids to a dense embedding of the sequence.
+
+    :param X: A tensor of token indexes with shape [batch_size, sequence_length, token_idx]
+    :param encoder: A TextEncoder object.
+    :param config: A config object, containing all parameters for the featurizer.
+    :param train: If this flag is true, dropout and losses are added to the graph.
+    :param reuse: Should reuse be set within this scope.
+    :return: A dict containing;
+        embed_weights: the word embedding matrix.
+        features: The output of the featurizer_final state.
+        sequence_features: The output of the featurizer at each timestep.
+    """
+    initial_shape = tf.shape(X)
+    X = tf.reshape(X, shape=tf.concat(([-1], initial_shape[-2:]), 0))
+    sequence_length = tf.shape(X)[1]
+    with tf.variable_scope("model/featurizer", reuse=reuse):
+        h = X # X is already embedded
+
+        clf_token = encoder.start_token
+
+        with tf.variable_scope("tcn_stack"):
+            representation = h
+            for layer_num in range(config.n_layer):
```

Comment:
Maybe we should initialize to have this block approximate some meaningful heuristic -- i.e. noisy mean or similar.